Homework 02
======================================================================

In this assignment we worked with data originally published in "A comprehensive comparison of RNA-Seq-based transcriptome analysis from reads to differential gene expression and cross-comparison with microarrays: a case study in _Saccharomyces cerevisiae_." by Nookaew et al (Nucleic Acids Res. 2012 Nov 1;40(20):10084-97. PMID 22965124). The article is available here: [doi: 10.1093/nar/gks804](http://dx.doi.org/10.1093%2Fnar%2Fgks804).

The authors used two different platforms -- microarrays and RNA-Seq -- to obtain gene expression data for yeast grown in two conditions: batch medium and chemostat. They then compared the results of differential expression analysis (DEA) across platforms and under various combinations of aligners and methods for identifying differential expression. We will do the same. Some of the DEA methods covered in this class were included in their analysis, `edgeR` and `DESeq`, while others were not, such as `limma` and, in particular, the `voom` function.

We will work with their data obtained from the NCBI SRA and GEO repositories. Because of some differences in pre-processing your results will differ from theirs.

## Q1) Microarray Analysis

The six samples in this study, 3 replicates for each of the two conditions, were analysed on the Affymetrix Yeast Genome Array 2.0 platform. We have already downloaded the raw CEL files from GEO and normalised them. The normalized data is saved in the file [`GSE37599-data.tsv`](../../examples/yeastPlatforms/data/GSE37599-data.tsv).

### a) (1pt) Load Microarray Data

Start by loading libraries (hidden):
```{r echo=FALSE}
suppressWarnings(library(ggplot2))
suppressWarnings(library(gplots))
suppressWarnings(library(DESeq))
library(lattice)
library(plyr)
library(knitr)
library(RColorBrewer)
library(hexbin)
library(preprocessCore)
library(limma)
library(yeast2.db)
```

Load the normalized data:
```{r}
gseDat <- read.table("GSE37599-data.tsv", row.names = 1, header = TRUE)
```
  
What are dimensions of the dataset?
```{r}
dim(gseDat)
```

In addition to reporting number of rows and columns, make it clear what rows and columns represent and how you're interpreting column names:

```{r}
str(gseDat)
head(gseDat)
```
I am interpreting the columns names as samples and rows names as probes. The intersetions are some measure of expression based on hybridization on the microarray. Column names (ie. samples) with the same prefix (b for batch medium and c for chemostat) have the same conditions.

### b) (2pt) Identify Sample Swap

The labels on two of the samples have been swapped, that is one of the batch samples has been labelled as chemostat and vice-versa. Produce the plots described below and explain how they allow you to identify the swapped samples.
  
i. (High volume) scatter plot matrix. 
```{r}
splom(gseDat, panel=panel.hexbinplot)
```
This plot suggests that c2 is swapped with b1 based on the correlation observed.

ii. A heatmap of the first 100 genes:
```{r}
#colour for heatmap
jGnBuFun <- colorRampPalette(brewer.pal(n = 9, "GnBu"))
heatmap.2(as.matrix(head(gseDat,n=100)), col = jGnBuFun, trace = "none")
```
Again, the plot suggests that c2 is swapped with b1 based on the groupings by the dendrogram and the expression paterns observed.

iii. The Pearson correlation of the samples and plot using a heatmap:
```{r}
#run correlation
corDat <- cor(gseDat)
heatmap.2(corDat, col = jGnBuFun, trace = "none")
```
Again, the plot suggests that c2 is swapped with b1 based on the correlation observed.

iv. Scatterplot the six data samples with respect to the first two principal components and label the samples.
```{r}
pcs <- prcomp(gseDat, center = F, scale = F)
pcsDat <- data.frame(pcs$rotation[,1:2])
pcsDat$samples <- rownames(pcsDat)

#with ggplot2
ggplot(pcsDat, aes(x = PC1, y = PC2, colour = samples)) + geom_point()

#with plot /w funtion labels
plot(pcs$rotation[,1:2])
text(pcs$rotation[,1:2], row.names(pcs$rotation[,1:2]), cex=0.6, pos=3)
```
Again, the plot suggests that c2 is swapped with b1 based on groupings.


### c) (2pt) Microarray Differential Expression

Fix the label swap identified in question 1b. We want to swap b1 <--> c2.
```{r}
#before
head(gseDat)
#after
gseDatFix <- gseDat
colnames(gseDatFix) <- c("c2","b2","b3","c1","b1","c3")
gseDatFix <- gseDatFix[,c(5,2,3,4,1,6)]
head(gseDatFix)
```

Revisit one or more elements of question 1b to sanity check before proceeding. 
scatter plot matrix. 
```{r}
splom(gseDatFix, panel=panel.hexbinplot)
```

A heatmap of the first 100 genes:
```{r}
heatmap.2(as.matrix(head(gseDatFix,n=100)), col = jGnBuFun, trace = "none")
```

Pearson correlation of the samples and plot using a heatmap:
```{r}
#run correlation
corDatFix <- cor(gseDatFix)
heatmap.2(corDatFix, col = jGnBuFun, trace = "none")
```

Now use this data to do a differential expression analysis with `limma`.

Package these results in a data frame with six columns:
* probe.id - The array probe id.
* gene.id - The id of the gene which the probe overlaps (see below).
* p.value - The raw p-value for the probe.
* q.value - The BH corrected p-value, aka the q-value.
* log.fc - The log fold change which is the column called "logFC" in the limma results table.
* test.stat - The test statistics which for limma is the moderated t statistic. This is the column called "t" in the limma results table.

```{r}
#create a design matrix
gseDes <- data.frame(sampleID = colnames(gseDatFix), replicate = c(1,2,3), type = c('batch','batch','batch','chemostat','chemostat','chemostat'))
rownames(gseDes) <- gseDes$sampleID
gseDes
gseDesMat <- model.matrix(~type, gseDes)
str(gseDesMat)

gseFit <- lmFit(gseDatFix, gseDesMat)
gseFitEB <- eBayes(gseFit)
colnames(coef(gseFitEB))

tTbl <- topTable(gseFitEB, number = Inf, coef = c("typechemostat"))

array.results <- data.frame(probe.id = rownames(tTbl), gene.id = unlist(mget(rownames(tTbl), yeast2ORF)), p.value = tTbl$P.Value, q.value = tTbl$adj.P.Val, log.fc = tTbl$logFC, test.stat =tTbl$t)
```

Remove any rows with probes which don't map to genes. You'll be able to find these because they will have `NA` as their gene id. Work with this data.frame to answer the questions below.
```{r}
array.results <- subset(array.results, !is.na(gene.id))
head(array.results)
```

i. How many probes did we start with and how many remain after removing probes without gene ids?

Probes started with:
```{r}
nrow(gseDatFix)
```

Probes currently:
```{r}
nrow(array.results)
```

ii. Illustrate the differential expression between the batch and the chemostat samples for the top hit (i.e., probe with the lowest p- or q-value).
```{r}
prepareData <- function(x, dat, des){
  miniDat <- subset(dat, rownames(dat) %in% x)
  miniDat <- data.frame(gExp = as.vector(t(as.matrix(miniDat))), probe.id = factor(rep(rownames(miniDat), each = ncol(miniDat)), levels = x))
  miniDat <- suppressWarnings(data.frame(des, miniDat))
  return(miniDat)
}

#get top hit
topHit <- array.results[order(array.results$p.value),][1,1]
topDat <- prepareData(topHit, gseDatFix, gseDes)
ggplot(topDat, aes(type, gExp)) + geom_violin() + geom_point(position = "jitter") + xlab("Method of Measurement") + ylab("Gene Expression")
```
As you can see the gene expression for the batch is much lower than the chemostat expression.


iii. How many probes are identified as differentially expressed at a false discovery rate (FDR) of 1e-5 (note: this is a FDR cutoff used in the original paper)?
```{r}
arrayTopGenes <- subset(array.results[order(array.results$p.value),], q.value < 1e-5)
nrow(arrayTopGenes)
```

iv. Save your results for later with `write.table()`.

```{r}
write.table(array.results, file = "GSE37599-diffExp.txt" ,row.names = TRUE, col.names = NA)
```

## Q2) RNA-Seq Analysis

We have aligned the RNA-Seq library using the [Stampy](http://www.well.ox.ac.uk/project-stampy) aligner and generated count data. The data file is available [here](../../examples/yeastPlatforms/data/stampy.counts.tsv). In this question you will use this data to do a differential expression analysis using different packages from Bioconductor.

### a) (1pt) Load RNA Count Data and Sanity Check

Load the count data using `read.table`; you will need to pass the arguments `header=TRUE` and `row.names=1`. 

```{r}
countDat <- read.table("stampy.counts.tsv" , header=TRUE, row.names=1)
```

i) What are dimensions of the dataset?
```{r}
dim(countDat)
```

In addition to reporting number of rows and columns, make it clear what rows and columns represent. What is the difference between the rows of this dataset versus rows of the array data in question 1a?
```{r}
str(countDat)
head(countDat)
```
I am interpreting the columns names as samples and rows names as genes (in contrast to 1a which were probes on a microarray). The intersetions are read counts. Column names (ie. samples) with the same prefix (b for batch medium and c for chemostat) have the same conditions.

ii) Do a sanity check to make sure there is no sample swap by plotting a heatmap of the sample correlations.
```{r}
#run correlation
corDat <- cor(countDat)
heatmap.2(corDat, col = jGnBuFun, trace = "none")
```
There doesn't seem to be a sample swap.

### b) (2pt) `edgeR` Differential Expression Analysis

Now you will use `edgeR` to identify differentially expressed genes between the batch medium vs. chemostat conditions.

i)  Recall that `edgeR` needs to estimate the dispersion parameter in the negative binomial model using an empirical Bayes method. Estimate the dispersion parameters using `estimateGLMCommonDisp`, `estimateGLMTrendedDisp` and `estimateGLMTagwiseDisp`. Plot the tagwise dispersion against log2-CPM (counts per million).  
```{r}
#load edge R
library(edgeR)
with(gseDes, table(type))
group <- factor(c(rep("1", 3), rep("2", 3)))
group
dge.glm <- DGEList(counts = countDat, group = group)
design <- model.matrix(~group)
design
#taken from seminar 7
dge.glm.com.disp <- estimateGLMCommonDisp(dge.glm, design, verbose = TRUE)
dge.glm.trend.disp <- estimateGLMTrendedDisp(dge.glm.com.disp, design)
dge.glm.tag.disp <- estimateGLMTagwiseDisp(dge.glm.trend.disp, design)
plotBCV(dge.glm.tag.disp)
```

ii)  Use the glm functionality of `edgeR`, i.e. use the `glmFit` function, to identify differentially expressed genes between conditions. 

Package these results in a data.frame called 'edger.results' with five columns:
* gene.id - The id of the gene which reads were aligned to.
* p.value - The raw p-value for the gene.
* q.value - The BH corrected p-value, aka the q-value.
* log.fc - The log fold change which is the column called "logFC" in the `edgeR` results table.
* test.stat - The test statistic, which for `edgeR` is a likelihood ratio. This is the column called "LR" in the `edgeR` results table.
```{r}
fit <- glmFit(dge.glm.tag.disp, design)
lrt <- glmLRT(fit, coef = 2)
tt.glm <- topTags(lrt, n = Inf)$table
edger.results <- data.frame(gene.id = rownames(tt.glm), p.value = tt.glm$PValue, q.value = tt.glm$FDR, log.fc = tt.glm$logFC, test.stat = tt.glm$LR )
```

Save your results for later with `write.table()` in file called `stampy.edger.results.tsv`.
```{r}
write.table(edger.results, "stampy.edger.results.tsv",row.names = TRUE, col.names = NA)
```

iii) How many genes are differentially expressed between conditions at a false discovery rate (FDR) of 1e-5?
```{r}
edgerTopGenes <- edger.results[edger.results$q.value < 1e-5, ]
nrow(edgerTopGenes)
```

iv) How many genes are differentially over-expressed in chemostat compared to batch medium samples at a false discovery rate (FDR) of 1e-5?
```{r}
de.glm <- decideTestsDGE(lrt, p = 1e-5, adjust = "fdr")
summary(de.glm)[3]
```
1515 genes are over-expressed.

### c) (2pt) `DESeq` Differential Expression Analysis

Now you will use `DESeq` to identify differentially expressed genes between the batch medium vs. chemostat conditions.

i)  `DESeq` also needs to estimate the dispersion. Use `estimateSizeFactors` and `estimateDispersions` to normalize the data. Plot the estimated dispersions against the mean normalized counts.

```{r}
deSeqDat <- newCountDataSet(countDat, group)
deSeqDat <- estimateSizeFactors(deSeqDat)
deSeqDat <- estimateDispersions(deSeqDat)
plotDispEsts(deSeqDat)
```

ii)  Use the negative binomial test of `DESeq`, i.e. use the `nbinomTest` function, to identify differentially expressed genes between conditions. Note that the output of this function does not return results ordered by p-values or logged fold-changes. You can manually reorder the results if you want (not required for this homework).

Package these results in a data.frame called 'deseq.results' with four columns:
* gene.id - The id of the gene which reads were aligned to.
* p.value - The raw p-value for the gene.
* q.value - The BH corrected p-value, aka the q-value.
* log.fc - The log fold change which is the column called "logFC" in the `edgeR` results table.
```{r}
deSeqResults <- nbinomTest(deSeqDat, levels(group)[1], levels(group)[2])
#reorder by pvalue
deSeqResults <- deSeqResults[with(deSeqResults, order(pval)), ]
deseq.results <- data.frame(gene.id = deSeqResults$id, p.value = deSeqResults$pval, q.value = deSeqResults$padj, log.fc = deSeqResults$log2FoldChange)
```

Save your results for later with `write.table()` in file called `stampy.deseq.results.tsv`.
```{r}
write.table(deseq.results, "stampy.deseq.results.tsv",row.names = TRUE, col.names = NA)
```

iii) How many genes are differentially expressed between conditions at a false discovery rate (FDR) of 1e-5?
```{r}
deseqTopGenes <- deseq.results[deseq.results$q.value < 1e-5, ]
nrow(deseqTopGenes)
```

iv) How many differentially expressed genes are identified by both 'edgeR' and 'DESeq'?
```{r}
sharedGenes <- intersect(edgerTopGenes$gene.id, deseqTopGenes$gene.id)
length(sharedGenes)
```

### d) (2pt) `voom` Differential Expression Analysis

Now you will use `voom+limma` to identify differentially expressed genes between the batch medium vs. chemostat conditions.

i)  `voom` normalizes the counts before it converts counts to log2-cpm. Use `calcNormFactors` to normalize counts.
```{r}
norm.factor <- calcNormFactors(countDat)
```

ii)  Use `voom' to convert count data into logged CPM data and then use 'limma' to identify differentially expressed genes between conditions. 

Package these results in a data.frame called 'voom.limma.results' with five columns:
* gene.id - The id of the gene which reads were aligned to.
* p.value - The raw p-value for the gene.
* q.value - The BH corrected p-value, aka the q-value.
* log.fc - The log fold change which is the column called "logFC" in the `edgeR` results table.
* test.stat - The test statistic, which is the column called "t".
```{r}
dat.voomed <- voom(countDat, design, plot = TRUE, lib.size = colSums(countDat) * norm.factor)
fit <- lmFit(dat.voomed, design)
fit <- eBayes(fit)
voomTTbl <- topTable(fit, n = Inf, coef = c("group2"))
limma.results <- data.frame(gene.id = rownames(voomTTbl), p.value = voomTTbl$P.Value, q.value = voomTTbl$adj.P.Val, log.fc = voomTTbl$logFC, test.stat <- voomTTbl$t)
```

Save your results for later with `write.table()` in file called `stampy.limma.results.tsv`.
```{r}
write.table(limma.results, "stampy.limma.results.tsv",row.names = TRUE, col.names = NA)
```


iii) How many genes are differentially expressed between conditions at a false discovery rate (FDR) of 1e-5?
```{r}
limmaTopGenes <- limma.results[limma.results$q.value < 1e-5, ]
nrow(limmaTopGenes)
```

iv)  What fraction of the genes identified using `voom+limma` are also found by `edger` and `DESeq` methods? For example if the DE analysis using `voom+limma` found 1000 genes and both `edgeR` and `DESeq`  found 500 of these, the fraction of genes found would be $\frac{500}{1000}=0.5$.
```{r}
sharedGenes3 <- intersect(limmaTopGenes$gene.id,sharedGenes)
length(sharedGenes3)/length(limmaTopGenes$gene.id)
```

### e) (3pt) Comparison of Differential Expression Analyses
 
Now that we have the results of the differential expression analysis performed by three popular methods, we are going to compare and illustrate the results.

i) In previous questions, we noticed that different methods identified different differentially expressed genes. Create a Venn diagram showing all genes identified as differentially expressed by `edgeR`, `DESeq`, and `voom+limma`. Check your if your answers to questions 2c-iv, and 2d-iv are correct.

```{r}
library(VennDiagram)

de.genes <- list(voom_limma = limmaTopGenes$gene.id, edgeR = edgerTopGenes$gene.id, DESeq = deseqTopGenes$gene.id)

plot.new()
venn.plot <- venn.diagram(de.genes, filename = NULL, fill = c("red", "blue", "green"))

grid.draw(venn.plot)
```
These values seem to be consitant from my previous results (2c-iv, and 2d-iv):
* DESeq: 22 + 384 + 1792 = 2198
* edgeR: 384 + 1792 + 2 + 491 = 2669
* voom+limma: 1792 + 2 = 1794
* DESeq and edgeR: 1792 + 384 = 2176
* voom+limma edgeR and DESeq: 1792 / 1794 = 0.9989

ii) Using the function `plotSmear` function from `edgeR`, you can look at a scatterplot of observed differential expression (y-axis) against overall abundance (x-axis), both axes logarithmically transformed -- to check that putative DE genes seem plausible. Create a smear plot. Within this plot, identify the set of genes which are differentially expressed at an FDR of 1e-5 using all three methods (i.e., the q-values estimated by `edgeR`, `DESeq`, and `voom` are below 1e-5). Explain how you interpret this plot. Do you see reasonable results?

```{r}
plotSmear(dge.glm, de.tags = sharedGenes3)
```
Points that are black are not siginifcantly differntially expressed, and points in red are differntially expressed by all methods. You can see almost a clear phase seperation of the black and red points reflecting our p-value cut off.

For the most part the results seem resonable. There are are a few points (presumbably from DESeq and voom) that are not labeled diffentially expressed (still black) despite being outside of the region of of signiicant (red) results. This too, however is to be expected as the methods compute q-values slightly differntly from eachother.

iii) There are two genes identified by `edgeR` and `voom+limma` but not by `DESeq`. Illustrate the logged counts of them. Compare the (log) counts of these two genes with those of two genes identified by the three methods (see example below)

```{r}
missingGenes <- setdiff(intersect(limmaTopGenes$gene.id, topGenes$gene.id), deseqTopGenes$gene.id)
(featureCounts <- countDat[missingGenes, ])
featureDat <- data.frame(gene.id = factor(rep(rownames(featureCounts), ncol(featureCounts))), 
    cond = factor(rep(gseDes$type, each = nrow(featureCounts))), log.count = log2(unlist(featureCounts)))
stripplot(gene.id ~ log.count, featureDat, groups = cond, auto.key = TRUE, jitter = TRUE)
```
TODO: DISCUSS!!

## Q3) Compare DEA results between RNA-Seq and array

In question 1, you performed a DEA of array data using `limma`. In question 2a, you performed a DEA of RNA-Seq data using `edgeR`, among other methods. In this question you will compare the results of those two analyses. 

i) Use a Venn diagram to display the overlap and non-overlap of the __genes__ identified as differentially expressed at an FDR of 1e-5 by these analyses.

```{r}
edgeRDEGenes <- list(RNA_Seq = edgerTopGenes$gene.id, Array = arrayTopGenes$gene.id)
plot.new()
venn.plot <- venn.diagram(edgeRDEGenes, filename = NULL, fill = c("red", "blue"), force.unique = TRUE)
grid.draw(venn.plot)
```

ii) As expected, more genes were identified as differentially expressed using RNA-Seq data. In this question, you will examine the difference between the q-values from both analyses by overlaying density plots of the q-values from each analysis.

To respond to this question, make two plots:
* One plot that includes the densities of q-values of the genes analyzed by both platforms (i.e., genes shared by both data frames)
* another plot that includes the densities of q-values of ALL genes analyzed by at least one of the platforms.
Make some observations about the strengths of these two platforms.

###Densities of q-values of the genes analyzed by both platforms and make some observations about the strengths of these two platforms:
For all genes by both platforms:
```{r}
bothDat <- intersect(edger.results$gene.id, array.results$gene.id)
arrayDatBoth <- array.results[array.results$gene.id %in% bothDat,]
arrayDatBoth$type <- "microarray"
rnaDatBoth <- edger.results[edger.results$gene.id %in% bothDat,]
rnaDatBoth$type <- "rna-seq"

intersectDat <- merge(rnaDatBoth,arrayDatBoth, all = TRUE)

p <- ggplot(intersectDat, aes(x=q.value, colour = type, fill = type)) + geom_density(alpha =0.5)
p
```

Basically we can see rna-seq seem to generate a similar density distribution, except near 0 (where it matters). More area seems to be concentrated near zero in RNA-seq and thus a higher probablity is concentrated at lower q-values. It seems to be able to call genes with higher confidence. This is likely due to limitations in microarray technology like lower dynamic range with respect to RNA-seq.

For only significant gene by both platforms:
```{r}
bothDat <- intersect(edgerTopGenes$gene.id, arrayTopGenes$gene.id)
arrayDatBoth <- arrayTopGenes[arrayTopGenes$gene.id %in% bothDat,]
arrayDatBoth$type <- "microarray"
rnaDatBoth <- edgerTopGenes[edgerTopGenes$gene.id %in% bothDat,]
rnaDatBoth$type <- "rna-seq"

intersectDat <- merge(rnaDatBoth,arrayDatBoth, all = TRUE)

#adjust for p-values of 0
intersectDat$q.value <- intersectDat$q.value + .Machine$double.xmin

p <- ggplot(intersectDat, aes(x=q.value, colour = type, fill = type)) + geom_density(alpha =0.5) + scale_x_log10()
p
```
Basically this shows that compared to RNA-seq, that most signicant micro array q-values are generally larger than siginicant RNA-seq values.

###Densities of q-values of ALL genes analyzed by at least one of the platforms and make some observations about the strengths of these two platforms:

For all genes by both platforms:
```{r}
allDat <- union(edger.results$gene.id, arrayTopGenes$gene.id)
arrayDatAll <- array.results[array.results$gene.id %in% allDat,]
arrayDatAll$type <- "microarray"
rnaDatAll <- edger.results[edger.results$gene.id %in% allDat,]
rnaDatAll$type <- "rna-seq"

unionDat <- merge(rnaDatAll,arrayDatAll, all = TRUE)

p <- ggplot(unionDat, aes(x=q.value, colour = type, fill = type)) + geom_density(alpha =0.5)
p
```

Again, we can see rna-seq seem to generate a similar density distribution, except near 0 (where it matters). More area seems to be concentrated near zero in RNA-seq and thus a higher probablity is concentrated at lower q-values. The only difference it that the effect seems much more pronouced, likely because RNA-seq is not dependant on what is hybridized to probes as it is in microarrays.

For only significant gene by both platforms:
```{r}
allDat <- union(edger.results$gene.id, arrayTopGenes$gene.id)
arrayDatAll <- arrayTopGenes[arrayTopGenes$gene.id %in% allDat,]
arrayDatAll$type <- "microarray"
rnaDatAll <- edgerTopGenes[edgerTopGenes$gene.id %in% allDat,]
rnaDatAll$type <- "rna-seq"

unionDat <- merge(rnaDatAll,arrayDatAll, all = TRUE)

#adjust for p-values of 0
unionDat$q.value <- unionDat$q.value + .Machine$double.xmin

p <- ggplot(unionDat, aes(x=q.value, colour = type, fill = type)) + geom_density(alpha =0.5) + scale_x_log10()
p
```
Again, this shows that compared to RNA-seq, that most signicant micro array q-values are generally larger than siginicant RNA-seq values.

iii) We provide a data set with array expression and count data for 5 interesting genes; below is also code to load it and a figure depicting it.

Consult the DEA results from your previous analyses for these genes. For each gene, state its status with respect to these analyses, i.e. where it falls in those Venn diagrams. Comment on the results and plots.

```{r}
jDat <- dget("featGenesData-q3-DPUT.txt")

ggplot(jDat, aes(x= arrayExp, y = probe.id, colour = cond)) + geom_point()
ggplot(jDat, aes(x= log.count, y = probe.id, colour = cond)) + geom_point()

compareDat <- data.frame(gene.id = unique(jDat$gene.id))

compareDat$array <- compareDat$gene.id %in% arrayTopGenes$gene.id
compareDat$limma <- compareDat$gene.id %in% limmaTopGenes$gene.id
compareDat$edger <- compareDat$gene.id %in% edgerTopGenes$gene.id
compareDat$deseq <- compareDat$gene.id %in% deseqTopGenes$gene.id
kable(compareDat, format = "markdown")
```

YDR345C: Significant in all

YDR384C: Significant in all

YBL025W: not significant in all in all

YCL042W: Significant in only RNA-seq Edge R computed set.

YGL209W: Significant in only Microarray computed set.

TODO DISCUSS
